From 9c94527f105e3b345f97804b2f8077361bce48f5 Mon Sep 17 00:00:00 2001
From: Vihang Karajgaonkar <vihang@cloudera.com>
Date: Wed, 15 Jun 2016 12:21:51 -0700
Subject: [PATCH 0624/1164] FIXUP: 82b36d0bcfe10d00e5b41c1f34b1881abd60fdb9
 (addendum)

Fixes for the post-commit jenkins failure. Backports part of changes from HIVE-11077 for QueryPlan.java which are needed to fix lineage2-3.q tests and TestHs2Hooks tests

Successfully ran the CDH5-Hive-1.1.0-Test-Patch jenkins job for testing this patch

Change-Id: I4400c2a2d99e9468874ab2f17945ea36edacac64
---
 .../java/org/apache/hadoop/hive/ql/QueryPlan.java  |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/ql/src/java/org/apache/hadoop/hive/ql/QueryPlan.java b/ql/src/java/org/apache/hadoop/hive/ql/QueryPlan.java
index f1fd890..ba2ab22 100644
--- a/ql/src/java/org/apache/hadoop/hive/ql/QueryPlan.java
+++ b/ql/src/java/org/apache/hadoop/hive/ql/QueryPlan.java
@@ -107,7 +107,7 @@
   private QueryProperties queryProperties;
 
   private transient Long queryStartTime;
-  private String operationName;
+  private HiveOperation operation;
 
   public QueryPlan() {
     this.reducerTimeStatsPerJobList = new ArrayList<ReducerTimeStatsPerJob>();
@@ -149,7 +149,7 @@ public QueryPlan(String queryString, BaseSemanticAnalyzer sem, Long startTime, S
     query.putToQueryAttributes("queryString", this.queryString);
     queryProperties = sem.getQueryProperties();
     queryStartTime = startTime;
-    this.operationName = operationName;
+    this.operation = operation;
     this.resultSchema = resultSchema;
   }
 
@@ -809,6 +809,6 @@ public void setQueryStartTime(Long queryStartTime) {
   }
 
   public String getOperationName() {
-    return operationName;
+    return operation == null ? null : operation.getOperationName();
   }
 }
-- 
1.7.9.5

